Modules/Sources/JetpackStats/Services/CSVExporter.swift

```diff
@@ -0,0 +1,143 @@
+import Foundation
+import SwiftUI
+
+protocol CSVExporterProtocol {
+    func generateCSV(from items: [any TopListItemProtocol], metric: SiteMetric) -> String
+}
+
+/// Exports stats data to CSV format following RFC 4180 standard
+struct CSVExporter: CSVExporterProtocol {
+    // RFC 4180: Use CRLF for line endings
+    static let lineEnding = "\r\n"
+
+    // Characters that require field escaping according to RFC 4180
+    static let charactersRequiringEscape = CharacterSet(charactersIn: ",\"\r\n")
+
+    func generateCSV(from items: [any TopListItemProtocol], metric: SiteMetric) -> String {
```

Comment:
Does the item need to be `TopListItemProtocol`? Can they just be `CSVExportable`?

If they do need to be both, you can declare the function as below to avoid casting:

```swift
func generateCSV<T: TopListItemProtocol + CSVExportable>(from items: [T], metric: SiteMetric) -> String
```